astyanax-recipes/src/main/java/com/netflix/astyanax/recipes/storage/ObjectReader.java

```diff
@@ -166,9 +174,15 @@ public void run() {
                     }
                     finally {
                         executor.shutdown();
-                        if (!executor.awaitTermination(maxWaitTimeInSeconds, TimeUnit.SECONDS)) {
-                            throw new Exception("Took too long to fetch object: " + objectName);
+                        while (!executor.isTerminated()) {
+                            try {
+                                if (!executor.awaitTermination(maxWaitTimeInSeconds, TimeUnit.SECONDS))
+                                    throw new Exception("Took too long to fetch object: " + objectName);
+                            } catch (InterruptedException e) {
+
+                            }
```

Comment:
thank you @jeremycnf  and @mfiguiere for quick reviews. 

The original intent for users to be able to pass an external executor to ObjectReader is performance related ( to avoid having zombie threads). Calling shutdown with external executor service would have defeated that purpose. I've made the changes that does not shutdown the external service.